TestFoundation/TestDate.swift

```diff
@@ -42,26 +42,27 @@ class TestDate : XCTestCase {
     
     func test_BasicConstruction() {
         let d = Date()
-        XCTAssertNotNil(d)
+        XCTAssert(d.timeIntervalSince1970 != 0)
+        XCTAssert(d.timeIntervalSinceReferenceDate != 0)
     }
 
     func test_descriptionWithLocale() {
         let d = NSDate(timeIntervalSince1970: 0)
         XCTAssertEqual(d.description(with: nil), "1970-01-01 00:00:00 +0000")
-        XCTAssertNotNil(d.description(with: Locale(identifier: "ja_JP")))
+        XCTAssertEqual(d.description(with: Locale(identifier: "en_US")), "Thursday, January 1, 1970 at 12:00:00 AM Greenwich Mean Time")
```

Comment:
We explicitly did not want to test the exact string output here because it can change from time to time due to formatting differences.

Reply:
Or perhaps we could use `en_US_POSIX` locale as it should not change it's formatting.

Reply:
I'm fine with checking that `isEmpty` is false

Reply:
Done